frontend/src/components/SubscriptionReport.js

```diff
@@ -50,8 +57,10 @@ const SubscriptionReport = ({ timeRange, url, id, refetchLinks }) => {
         : undefined,
     [data]
   );
-  if (failureCount < 1 && (!data || isLoading)) return <Spinner />;
-  if (failureCount >= 1 && (!data || isLoading))
+
+  if ((failureCount < 1 && (!data || isLoading)) || isFetching)
```

Comment:
If you trigger `refetch()` on usePresignedURL then `isLoading` will toggle in to true (Im not sure if you need `isFetching` in conditions here)

Reply:
Mainly need because we need some was change on here because without it redraw not calling and isLoading won't work manly.